adview-manager/src/helpers.rs

```diff
@@ -231,6 +236,185 @@ mod test {
+
+            let anchor_selector = Selector::parse("div>a").unwrap();
+            let anchor = fragment
+                .select(&anchor_selector)
+                .next()
+                .expect("There should be an anchor");
+
+            let video_selector = Selector::parse("div>a>video").unwrap();
+            let video = fragment
+                .select(&video_selector)
+                .next()
+                .expect("There should be a video");
+
+            assert_eq!("div", div.value().name());
+
+            assert_eq!("a", anchor.value().name());
+            assert_eq!(
+                Some("https://ambire.com?utm_source=AdEx+(https://adex.network)"),
+                anchor.value().attr("href")
+            );
+
+            assert_eq!("video", video.value().name());
+            assert_eq!(Some("480"), video.value().attr("width"));
+            assert_eq!(Some("480"), video.value().attr("height"));
+        }
+        // Test for another size
+        {
+            let unit = get_unit_html(Some(size_2), &video_unit, "https://adex.network", "", "");
+            let fragment = Html::parse_fragment(&unit);
+
+            let video_selector = Selector::parse("div>a>video").unwrap();
+            let video = fragment
+                .select(&video_selector)
+                .next()
+                .expect("There should be a video");
+
+            assert_eq!("video", video.value().name());
+            assert_eq!(Some("920"), video.value().attr("width"));
+            assert_eq!(Some("160"), video.value().attr("height"));
+        }
+
+        // Test for image ad_unit
+        {
+            let unit = get_unit_html(Some(size_1), &image_unit, "https://adex.network", "", "");
+            let fragment = Html::parse_fragment(&unit);
+
+            let image_selector = Selector::parse("div>a>*").unwrap();
+            let image = fragment
+                .select(&image_selector)
+                .next()
+                .expect("There should be an image");
+
+            assert_eq!("img", image.value().name());
+        }
+
+        // Test for another hostname
+        {
+            let unit = get_unit_html(
+                Some(size_1),
+                &video_unit,
+                "https://adsense.google.com",
+                "",
+                "",
+            );
+            let fragment = Html::parse_fragment(&unit);
+
+            let anchor_selector = Selector::parse("div>a").unwrap();
+            let anchor = fragment
+                .select(&anchor_selector)
+                .next()
+                .expect("There should be a second anchor");
+
+            assert_eq!("a", anchor.value().name());
+            assert_eq!(
+                Some("https://ambire.com?utm_source=AdEx+(https://adsense.google.com)"),
+                anchor.value().attr("href")
+            );
+        }
+    }
+
+    #[test]
+    fn getting_unit_html_with_events() {
+        let whitelisted_tokens = DEFAULT_TOKENS.clone();
```

Comment:
For the whitelisted tokens we can use `GANACHE_CONFIG` instead:

```
let whitelisted_tokens = config
        .chains
        .values()
        .flat_map(|chain| chain.tokens.values().map(|token| token.address))
        .collect::<Vec<_>>()
```

We can add a separate static in a different branch so I can use it too.